bitcoin/core/key.py

```diff
@@ -67,7 +192,10 @@ class CECKey:
     POINT_CONVERSION_UNCOMPRESSED = 4
 
     def __init__(self):
-        self.k = _ssl.EC_KEY_new_by_curve_name(_NID_secp256k1)
+        # WTF: not sure why the cast is necessary here, but we get back an
+        # int without it, even though we set
+        # _ssl.EC_KEY_new_by_curve_name.restype above
+        self.k = ctypes.c_void_p(_ssl.EC_KEY_new_by_curve_name(_NID_secp256k1))
```

Comment:
... if we don't have this "cast", `EC_KEY_new_by_curve_name` seems to return an `int`<sup>1</sup>, which means the various invocations of `ctypes.byref(self.k)` will throw an exception. The "cast" fixes this, but (at least in theory) it shouldn't be necessary.

--

<sup>1</sup> Although it appears that on 64-bit machines the `int` does have a high word, so that's at least something. :grimacing: